scripts/commit-tree.js

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env node
+
+/*!
+ * commit-tree.js - git commit sha512 tree hashes
+ * Copyright (c) 2018, the bcoin developers (MIT License).
+ * https://github.com/bcoin-org/bcoin
+ *
+ * Based on:
+ * - https://github.com/bitcoin/bitcoin/commit/fa89670d34ac7839e7e2fe6f59fdfd6cc76c4483
+ * - https://github.com/bitcoin/bitcoin/pull/9871
+ *
+ * Modify git commit message with Tree-SHA512:
+ * ```
+ * ./scripts/commit-tree.js <message>
+ * ```
+ *
+ * Verify tree hashes by running:
+ * ```
+ * git ls-tree --full-tree -r --name-only HEAD | LANG=C sort \
+ *   | xargs -n 1 sha512sum | sha512sum
+ * ```
+ */
+
+'use strict';
+
+const path = require('path');
+const {createReadStream} = require('fs');
+const {execFile} = require('child_process');
+const SHA512 = require('bcrypto/lib/sha512');
+
+process.title = 'bcoin-commit-tree';
+
+async function gitTreeList() {
+  return new Promise((resolve, reject) => {
+    execFile(
+      'git', ['ls-tree', '--full-tree', '-r', '--name-only', 'HEAD'],
+      (err, stdout) => {
+        if (err)
+          reject(err);
+        resolve(stdout.trim().split('\n').sort());
+      });
+  });
+}
+
+async function hashFile(file) {
+  const ctx = new SHA512().init();
+
+  return new Promise((resolve, reject) => {
+    createReadStream(file, {flags: 'r'})
+      .on('error', reject)
+      .on('data', data => ctx.update(data))
+      .on('end', () => resolve(ctx.final()));
+  });
+}
+
+async function gitTreeHash() {
+  const files = await gitTreeList();
+  const ctx = new SHA512().init();
+
+  while (files.length > 0) {
+    const f = files.shift();
+    const hash = await hashFile(f);
+    ctx.update(Buffer.from(hash.toString('hex'), 'utf8'));
+    ctx.update(Buffer.from(`  ${f}\n`, 'utf8'));
+  }
+
+  return ctx.final();
+}
+
+async function gitCommitMessage(message, signed, amend) {
```

Comment:
should `message` be initialized to an empty string? I got a message that was literally the string `undefined` running without a message arg. See commit 91e9d9a [here](https://github.com/bpanel-org/bpanel/pull/650)

Reply:
Yeah, or give an error w/ missing a missing message.

Reply:
The message is initialized as empty string at line 94.

Reply:
Yeah I saw that, but it still got an undefined string when I ran `npm run commit-tree`

Reply:
Hmm... could be 97?

Reply:
I get the same.